app/src/common/shared/org/mozilla/vrbrowser/utils/UrlUtils.java

```diff
@@ -0,0 +1,160 @@
+/* -*- Mode: Java; c-basic-offset: 4; tab-width: 4; indent-tabs-mode: nil; -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.vrbrowser.utils;
+
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+import android.webkit.URLUtil;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+
+// This class refers from mozilla-mobile/focus-android
+public class UrlUtils {
+    public static String normalize(@NonNull String input) {
+        String trimmedInput = input.trim();
+        Uri uri = Uri.parse(trimmedInput);
+
+        if (TextUtils.isEmpty(uri.getScheme())) {
+            uri = Uri.parse("http://" + trimmedInput);
+        }
+
+        return uri.toString();
+    }
+
+    /**
+     * Is the given string a URL or should we perform a search?
+     *
+     * TODO: This is a super simple and probably stupid implementation.
+     */
+    public static boolean isUrl(String url) {
+        String trimmedUrl = url.trim();
+        if (trimmedUrl.contains(" ")) {
```

Comment:
this most certainly could be triggered as a false positive by a space in unencoded hash or query-string values

Reply:
I have removed it at my new update. Thanks!